### PR TITLE
Replace mutations for vtex id API

### PR DESCRIPTION
**Pull request by anitavincent**
#### What is the purpose of this pull request?
Start using the VTEX ID API to perform sendToken, loginWithPassword, validateToken, setNewPassword and OAuthRedirect operations, replacing the mutations.

#### What problem is this solving?
Stop setting cookies through graphql and start using the standard vtexid API for authentication operations.

#### How should this be manually tested?
All login operations should behave like before

#### Screenshots or example usage
Just try to log in or set a new password, the behavior should be the same
https://anita--storecomponents.myvtex.com/

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Comment by vitoriaHeliane** [comment]
How can we test it?

**Comment by vitoriaHeliane** [comment]
Remember to assign the PR.

File: react/components/EmailAndPassword.js [hunk 1]
```diff
@@ -165,48 +155,85 @@ class EmailAndPassword extends Component {
...
       >
         <div className="vtex-login__form-link-container flex justify-center ph0 mt4">
-          <a href="" className="link dim c-link" onClick={e => this.handleCreatePassword(e)}>
-            <span className="f7">{translate('login.notHaveAccount', intl)}</span>
+          <a
+            href=""
+            className="link dim c-link"
+            onClick={e => this.handleCreatePassword(e)}
+          >
+            <span className="f7">
```
Comment (Klynger) [inline react/components/EmailAndPassword.js]: I think this `f7` class should be replace for a design token

File: react/components/OAuth.js [hunk 2]
```diff
@@ -1,61 +1,44 @@
...
-        </Button>
+        <AuthService.OAuthRedirect useNewSession provider={provider} >
+          {({ loading, action: redirectToOAuthPage }) => (
+            <Button
+              isLoading={loading}
+              variation="tertiary"
+              onClick={redirectToOAuthPage}
+            >
+              {children}
+              <span className="f6 vtex-login__oauth-label relative normal">
```
Comment (Klynger) [inline react/components/OAuth.js]: Design token for `f6` too

File: react/components/RecoveryPassword.js [hunk 3]
```diff
@@ -127,40 +131,61 @@ class RecoveryPassword extends Component {
...
                 variation="tertiary"
                 size="small"
                 onClick={() => onStateChange({ step: previous })}
               >
-                <span className="vtex-login__back-icon"><IconArrowBack size={10} color="#368DF7" /></span>
-                <span className="f7 ml2">{translate('login.goBack', intl)}</span>
+                <span className="vtex-login__back-icon">
+                  <IconArrowBack size={10} color="#368DF7" />
+                </span>
+                <span className="f7 ml2">
```
Comment (Klynger) [inline react/components/RecoveryPassword.js]: Design tokens

File: react/components/RecoveryPassword.js [hunk 4]
```diff
@@ -127,40 +131,61 @@ class RecoveryPassword extends Component {
...
+                }) => (
+                  <Button
+                    variation="primary"
+                    size="small"
+                    type="submit"
+                    onClick={e => this.handleOnSubmit(e, password, token, setPassword)}
+                    isLoading={loading}
+                    disabled={!validatePassword(password)}
+                  >
+                    <span className="f7">
```
Comment (Klynger) [inline react/components/RecoveryPassword.js]: design tokens

**Comment by rnbez** [comment]
> How can we test it?

@vitoriaHeliane You can test it by linking the app and trying to login using each of the identity providers available. All of them (password, accesskey, google and facebook) should behave just like before, but with AJAX instead of GraphQL mutations.

File: react/components/EmailAndPassword.js (hunk 1 above)
Comment (anitavincent) [reply react/components/EmailAndPassword.js]: Done

File: react/LoginContent.js [hunk 5]
```diff
@@ -307,26 +302,29 @@ class LoginContent extends Component {
...
-        <Transition
-          keys={(!profile && this.shouldRenderLoginOptions && !loading) ? ['children'] : []}
-          from={{ opacity: 0, transform: 'translateX(-50%)' }}
-          enter={{ opacity: 1, transform: 'translateX(0%)' }}
-          leave={{ display: 'none' }}
-        >
-          {(!profile && this.shouldRenderLoginOptions && !loading) ? [this.renderChildren] : []}
-        </Transition>
-        <div className={formClassName}>
+      <AuthState store="store">
```
Comment (felippenardi) [inline react/LoginContent.js]:
```suggestion
      <AuthState scope="store">
```

File: react/components/EmailAndPassword.js [hunk 6]
```diff
@@ -74,37 +58,25 @@ class EmailAndPassword extends Component {
...
-  handleSuccess = status => {
-    status === 'Success' && this.props.loginCallback()
+  handleSuccess = () => {
+    this.props.loginCallback()
   }
 
-  handleWrongCredentials = status => {
-    status === 'WrongCredentials' && this.setState({ isWrongCredentials: true, isLoading: false })
+  handleFailure = err => {
+    err.authStatus === 'WrongCredentials'
```
Comment (luciannojunior) [inline react/components/EmailAndPassword.js]: Use a `switch` instead

File: react/components/EmailAndPassword.js [hunk 7]
```diff
@@ -165,48 +155,85 @@ class EmailAndPassword extends Component {
...
+              <div className="vtex-login__back-button">
+                <Button
+                  variation="tertiary"
+                  size="small"
+                  onClick={() =>
+                    onStateChange({ step: previous, password: '' })
+                  }
+                >
+                  <span className="vtex-login__back-icon">
+                    <IconArrowBack size={10} color="#368DF7" />
```
Comment (luciannojunior) [inline react/components/EmailAndPassword.js]: Use design tokens

**Comment by luciannojunior** [comment]
The first click on Login button for email/password mode doesn't login, just hides the hints. Is this the expected behavior?

**Comment by luciannojunior** [comment]
Please, @anitavincent, verify the 'Forgot Password' flow.

This error happens, and the label for the access code in portuguese is mispelled.
image

File: react/components/EmailAndPassword.js (hunk 6 above)
Comment (rnbez) [reply react/components/EmailAndPassword.js]: @luciannojunior why should we use a `switch` with a single `case`?

**Comment by rnbez** [comment]
@luciannojunior The issues you pointed were fixed.

**Comment by anitavincent** [comment]
> The first click on Login button for email/password mode doesn't login, just hides the hints. Is this the expected behavior?

@luciannojunior, I don't believe that's the expected behaviour, I found that and many other minor mistakes while working on this PR, behaviour that was already there beforehand. Like misspelled variables, deprecation warnings, missing props, weird behaviour (like this and some other details), misspelled i18n text, linting issues. I started fixing some of these as you can see on the *fix* commits, but the PR started getting too off-topic. 

I decided to focus on working on the mutations only for this PR, and intend to work on the other issues on a future pull request.

> > @luciannojunior The issues you pointed were fixed.
> 
> Ok, @rnbez. Are you now the assignee on this PR? If so, please change its **Assignee**.
> 
> And also, about `The first click on Login button for email/password mode doesn't login, just hides the hints. Is this the expected behavior?`, I noticed this problem is presented in the current version of login app... I think it would be more appropriate to discuss in another matter.
> 
> So, from me, there's just the changes requested in the code 😄

@luciannojunior, the other issues you mentioned were already there too, I was planing on fixing them on my next PR, I had already fixed some on this pr (fix commits). And now @rnbez fixed the rest.

I'll make a new PR on the future, changing some more stuff. I'll create issues for what's missing.

**Comment by rnbez** [comment]
> Ok, @rnbez. Are you now the assignee on this PR? If so, please change its Assignee.
@anitavincent and I are working closely on this. I'm the developer of VTEX ID.

I agree that the behavior caused by the broken PasswordInput component should be discussed in another PR. The whole component should be replaced by the one in Styleguide.

As for the rest, I back what @anitavincent stated.

**Comment by rnbez** [comment]
> Ok, @rnbez. Are you now the assignee on this PR? If so, please change its Assignee.
@luciannojunior @anitavincent and I are working closely on this PR. I'm the developer of VTEX ID.

I agree that the behavior caused by the broken PasswordInput component should be discussed in another PR. The whole component should be replaced by the one in Styleguide.

As for the rest, I back what @anitavincent stated.

File: package.json [hunk 8]
```diff
@@ -0,0 +1,10 @@
+{
+  "devDependencies": {
```
Comment (brunojdo) [inline package.json]: Why this package.json is necessary? 🤔

File: react/components/EmailAndPassword.js (hunk 6 above)
Comment (brunojdo) [reply react/components/EmailAndPassword.js]: This approach seems ok to me. I will dismiss this issue

**Comment by Klynger** [comment]
> @Klynger Rafael, may I suggest we keep the design tokens fixes in a dedicated PR? This PR is not adding incorrect classes, it is just reusing what was there before.
> 
> Let's keep this PR focused in Replace mutations for vtex id API?
> Here is the issue for that: #84

Ok

File: package.json (hunk 8 above)
Comment (rnbez) [reply package.json]: for linting

**Pull request merged**